js/base/Exchange.js

```diff
@@ -2190,6 +2190,24 @@ module.exports = class Exchange {
         return [ type, params ];
     }
 
+    handleMethodAndParams (methodName, market = undefined, defaultMethod, allowedMethods, params = {} ) {
+        const methodOptions = this.safeValue (this.options, methodName);
+        let chosenMethod = defaultMethod;
+        if (methodOptions !== undefined) {
+            // if defined with new way: 'fetchXyz': { 'method': 'privateGetMarketOrders'}
+            chosenMethod = this.safeString2 (methodOptions, 'defaultMethod', 'method', chosenMethod);
+        } else {
+            // if defined with old way: 'fetchXyz': 'privateGetMarketOrders'
+            chosenMethod = this.safeValue (this.options, methodName + 'Method', chosenMethod);
+        }
+        chosenMethod = this.safeString (params, 'method', chosenMethod);
+        if (!allowedMethods.includes (chosenMethod)) {
```

Comment:
We must avoid cycling over an array to find a string-match O(n).
Instead we should do a O(logn) lookup in a dict.